text/0005-application-formalization.md

```diff
@@ -0,0 +1,191 @@
+- Start Date: 10/09/2023
+- Contributors:
+  - [Craig McNally](cmcnally@ebsco.com)
+  - [Vince Bareau](vbareau@ebsco.com)
+- RFC PRs:
+  - DRAFT REVIEW: https://github.com/folio-org/rfcs/pull/14
+  - PRELIMINARY REVIEW: TBD
+  - PUBLIC REVIEW: TBD
+  - FINAL REVIEW: TBD
+- Outcome: <!-- Leave this blank. Will eventually be either ACCEPTED or REJECTED -->
+
+# Application Formalization
+
+## Summary
+It is proposed that an Application be defined as the minimal but complete set of elements which together are intended to deliver a specific solution to Folio. The intention is to group together all the closely related components which will need to evolve together, but independently of other parts (i.e. other applications) of Folio.
+
+## Motivation
+The existing Folio project lacks a formal definition of an Application structure. A consequence is that Folio is stuck in a mode of monolithic releases which are ever increasing in size.  There have also been repeated complaints that the setup and management of Folio can be difficult due to the number modules involved.
+
+## Scope
+* Application composition
+* Application versioning
+* Application dependencies
+* Application definition/descriptors
+* Application management is ***out of scope***
+* Application-specific UI bundles is ***out of scope***
+* How movement of modules between Applications will work is ***out of scope***
```

Comment:
There are ~30 questions raised in the TC doc specific to application formalization.  A few are already listed here, or below in Unresolved Questions, or as other comments I've made here.  I'm itemizing the rest here (as I parse it).  

I don't think it's necessary to **answer**  them at the preliminary review stage.  I'm not clear though if we need to discuss at this stage whether they are in **scope**.

- Is the application formalization desirable without the platform formalization? 
- Are there product requirements addressed by these changes?
- With this new definition of “application”, what should we call the top-level menu items in the FOLIO UI (p.24), i.e. “Check In”? 
- What is the impact of defining applications on flower releases? 
  - How does that affect the marketing of FOLIO?  I.e. current press releases at the flower release milestones.  
  - If we are struggling with monolithic releases, why make a change to the FOLIO architecture rather than to the release processes? 
- If “evolution” (p.4) moves to the application level does governance move to the application level? 
  - What was meant by “evolution” (p.4)?
  - Would the Product Council evaluate (the functionality of) new applications instead of modules? 
  - Would the Product Council evaluate (the functionality of) new modules that are added to existing applications? 
  - Would the Technical Council review submitted applications? 
  - Would the Technical Council continue to review new modules?  What if the new modules are packaged into existing applications?
- How is BugFest testing impacted by a different release cycle for individual applications?